include/pybind11/cast.h

```diff
@@ -432,7 +432,7 @@ PYBIND11_NOINLINE inline std::string error_string() {
 
 #if !defined(PYPY_VERSION)
     if (scope.trace) {
-        PyTracebackObject *trace = (PyTracebackObject *) scope.trace;
+        auto *trace = (PyTracebackObject *) scope.trace;
```

Comment:
`auto *trace` or `auto trace`? (Not sure what's most common.)

Reply:
I think the star to remind that it's a pointer? Especially with the star aligned on the right (which some people really hate). Not sure either though.

Reply:
Probably. It obviously works, ofc, but you're still kind of repeating part of the type (just not the name part), so it's a weird correction to me...

No need to change it, though, as far as I'm concerned. Just curious on the conventions here.

> (which some people really hate)

I'm a (big) fan, btw ;-)

Reply:
I think there's a fully qualified auto check that adds stars and consts to auto if they apply (IIRC).

Reply:
I understand both * camps. The argument for left star is that it's really part of the type, not the name, and right star is that is (sadly, IMO) how the C syntax decided to do multiple definitions in one line. Which left star camp believes should be avoided, with one def per line always.

Reply:
> I think there's a fully qualified auto check that adds stars and consts to auto if they apply (IIRC).

Not convinced we always want that, actually. I don't see why `*` would be treated specially. Why would `int *` become `auto *`, but `std::unique_ptr<int>` become `auto`?

Reply:
> Which left star camp believes should be avoided, with one def per line always.

I actually agree, there (almost always, at least). And yet `T *x` is what I'm used to, and really not harder to read, as far as I can tell, vs.  `T* x`...